ui/app/components/auth-config-form/options.js

```diff
@@ -8,61 +8,63 @@ import AuthConfigComponent from './config';
 
 /**
  * @module AuthConfigForm/Options
  * The `AuthConfigForm/Options` is options portion of the auth config form.
  *
  * @example
  * ```js
- * {{auth-config-form/options model.model}}
+ * <AuthConfigForm::Options @modle={{this.args.model.model}} />
  * ```
  *
  * @property model=null {DS.Model} - The corresponding auth model that is being configured.
  *
  */
 
-export default AuthConfigComponent.extend({
-  flashMessages: service(),
-  router: service(),
+export default class AuthConfigOptions extends AuthConfigComponent {
+  @service flashMessages;
+  @service router;
 
-  saveModel: task(
-    waitFor(function* () {
-      const data = this.model.config.serialize();
-      data.description = this.model.description;
-      data.user_lockout_config = {};
+  @tracked errorMessage;
 
-      // token_type should not be tuneable for the token auth method.
-      if (this.model.methodType === 'token') {
-        delete data.token_type;
-      }
+  @task
+  @waitFor
+  *saveModel(evt) {
+    evt.preventDefault();
+    this.error = null;
+    const data = this.args.model.config.serialize();
+    data.description = this.args.model.description;
+    data.user_lockout_config = {};
+
+    // token_type should not be tuneable for the token auth method.
+    if (this.args.model.methodType === 'token') {
+      delete data.token_type;
+    }
 
-      this.model.userLockoutConfig.apiParams.forEach((attr) => {
-        if (Object.keys(data).includes(attr)) {
-          data.user_lockout_config[attr] = data[attr];
-          delete data[attr];
-        }
-      });
+    this.args.model.userLockoutConfig.apiParams.forEach((attr) => {
+      if (Object.keys(data).includes(attr)) {
+        data.user_lockout_config[attr] = data[attr];
+        delete data[attr];
+      }
+    });
 
-      try {
-        yield this.model.tune(data);
-      } catch (err) {
-        // AdapterErrors are handled by the error-message component
-        // in the form
-        if (err instanceof AdapterError === false) {
-          throw err;
-        }
-        // because we're not calling model.save the model never updates with
-        // the error.  Forcing the error message by manually setting the errorMessage
-        try {
-          this.model.set('errorMessage', err.errors?.join(','));
-        } catch {
-          // do nothing
-        }
-        return;
+    try {
+      yield this.args.model.tune(data);
+    } catch (err) {
+      // AdapterErrors are handled by the error-message component
+      // in the form
+      if (err instanceof AdapterError === false) {
+        throw err;
       }
-      this.router.transitionTo('vault.cluster.access.methods').followRedirects();
-      this.flashMessages.success('The configuration was saved successfully.');
-    })
-  ),
-});
+      // because we're not calling model.save the model never updates with
+      // the error. Set it on the component instead.
+      this.errorMessage = errorMessage(err);
```

Comment:
If we're throwing errors above and adapter errors are handled by MessageError in the form - what error could make it here to `errorMessage` ?

Reply:
Any other kind of error -- eg. error thrown manually from the adapter method, or a javascript parsing error

Reply:
Coming back to this -- the Adapter errors are making it to this line (line 57 is `=== false`) so the other ones that I listed above would be thrown. Not written very clearly so I might rewrite!